template/koa-rest-api/src/framework/validation.test.ts

```diff
@@ -43,13 +43,21 @@ describe('validate', () => {
       .expect(422)
       .expect(({ body }) =>
         expect(body).toMatchInlineSnapshot(`
-          {
-            "invalidFields": {
-              "/id": "Expected string, received null",
-            },
-            "message": "Input validation failed",
-          }
-        `),
+{
+  "invalidFields": {
+    "/id_0": [
```

Comment:
For what it's worth, the initial concept here was to return [JSON Pointers](https://datatracker.ietf.org/doc/html/rfc6901) to the specific part of the document that failed validation.

This change breaks that convention as it introduces its own syntax for unions, and it means that any clients looking to parse the response would have to understand said syntax. That said, I'm fine with moving ahead with this if we feel it's worthwhile, e.g. because in the typical case we just want to be able to tell what went wrong when reading the error message in its entirety, and because it would be rather unusual to have an organic field name that ends with `_X` that this could be confused with.

Reply:
Some confusion may happen if we want to address the order of union error. 
For example, following JSON Pointers with union error, we may have something like `/id/0`, which may confused with [array schema](https://github.com/colinhacks/zod?tab=readme-ov-file#arrays).

We can probably have an unusual postfix, like `_union0` to clear up confusion at a certain level, although it breaks the convention too.